salt/cloud/clouds/vmware.py

```diff
@@ -770,27 +770,70 @@ def _wait_for_vmware_tools(vm_ref, max_wait):
     return False
 
 
+def _valid_ip(ip_address):
```

Comment:
This is used for validation  for valid IP. Here added condition to check 169.254.x.x (private networking address) to Windows VM as this range for IP is assign to windows VM before setting actual IP.